gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py

```diff
@@ -2016,6 +2016,33 @@ def impl(context, filename, contain, output):
         if (not valuesShouldExist) and (output in actual):
             raise Exception('File %s on host %s contains "%s"' % (filepath, host, output))
 
+@given('verify that the path "{filename}" in each segment data directory does not exist')
+@then('verify that the path "{filename}" in each segment data directory does not exist')
+def impl(context, filename):
+    try:
+        with dbconn.connect(dbconn.DbURL(dbname='template1'), unsetSearchPath=False) as conn:
+            curs = dbconn.execSQL(conn, "SELECT hostname, datadir FROM gp_segment_configuration WHERE content > -1;")
+            result = curs.fetchall()
+            segment_info = [(result[s][0], result[s][1]) for s in range(len(result))]
+    except Exception as e:
+        raise Exception("Could not retrieve segment information: %s" % e.message)
+
+    for info in segment_info:
+        host, datadir = info
+        filepath = os.path.join(datadir, filename)
+        cmd_str = 'ls -la %s | grep %s | wc -l' % (datadir, filename)
```

Comment:
This approach may lead to false positive results:

```
givi@samsung-ivanov:~$ ls -la /tmp/2 | grep foo | wc -l
1
givi@samsung-ivanov:~$ ls /tmp/2
foobar
```

I think that check exact path is better:

```
givi@samsung-ivanov:~$ ls -la /tmp/2/foo
ls: cannot access '/tmp/2/foo': No such file or directory
```

Reply:
bash check directory exists
```sh
test -d /etc && echo true || echo false
true
```
```sh
test -d /etc1 && echo true || echo false
false
```